### DISCUSSION TITLE
How to add Sampler, LoRa

### DISCUSSION THREAD

**Discussion by steto123** [seruva19/kubin#135] [Q&A] [answered]
i am not so familiar with programming and such things. This is the reason for my question. How can i add other samplers, LoRa's, upscaler and other things to the app? Also there is a Aspect ratio box with only none as parameter. How can i add other ratio's like 16:9 or 4:3 to this (i do this manually at this time)? 

Many thanks for this nice work

**Comment by seruva19** [comment, marked as answer]
Appreciate your feedback!

Unfortunately, due to a lack of free time at the moment, I do not expect making a lot of updates until September.
Regarding your points:
1) Samplers: I've been experimenting with different samplers (schedulers) from diffusers library; however, I've been encountering a lot of errors and distorted images when using anything other than the default sampler. Because of this, I disabled option for sampler selection. That's a temporary measure. It will be unlocked once the issues have been resolved.
2) LoRA: To use LoRA, one must train it ;) At present, I am trying to train some LoRAs myself. As soon as I get noteworthy results, I will surely share it. If you are interested in training LoRA on your own, I can give you some instructions about on how to do so (all tools for training can be found in 'Training' tab of the app). For now, it's uncharted territory, so I have no idea what is most effective way to train LoRA for Kandinsky :)
3) Upscalers: I've started recently reworking upscaling extension. Planning to at least add more upscalers alongside RealESRGAN. Ultimately, I aim to achieve functionality similar to A1111's, enabling user to simply place a .pth file into upscalers directory and use it without any extra steps.
4) Aspect ratio: well, I started implementing it, but got carried away with another problems ;) so postponed this feature for a later time. This is why it's currently locked. 

I understand that lack of documentation also makes it challenging to use the app and eventually I will add some basic documentation. However, right now, I don't have time for this neither :(

**Comment by steto123** [comment]
many thanks,

if I can help, let me know. Unfortunately I'm only a beginner in php and python programming but I could do the one or other test.
